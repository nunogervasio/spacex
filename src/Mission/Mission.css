.container {
  display: flex;
  flex-direction: column;
  align-items: center;
}
@media (min-width: 576px) {
  .container {
    flex-direction: row;
    flex-wrap: wrap;
    justify-content: center;
    width: 90%;
    margin: 0 auto;
  }
}
