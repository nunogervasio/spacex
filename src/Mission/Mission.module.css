.container {
  display: flex;
  flex-direction: column;
  align-items: center;
}
@media (min-width: 576px) {
  .container {
    flex-direction: row;
    justify-content: center;
    width: 90%;
    margin: 0 auto;
  }
}
