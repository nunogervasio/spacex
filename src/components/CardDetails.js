import React from "react";
import PropTypes from "prop-types";
import youtubeImg from "../img/youtube-1.svg";
import wikiImg from "../img/wiki-1.svg";
const CardDetails = ({
  patch_url,
  number,
  name,
  date,
  site,
  rocketName,
  rocketType,
  details,
  youtube_url,
  wiki_url
}) => {
  return (
    <div className="main-card-details">
      <div className="card-img-details">
        <img src={patch_url} className="patch" alt="mission patch" />
      </div>

      <div className="card-details-details">
        <p>
          <span className="flight-name">{name}</span>
          <br />
          <span className="card-location">Flight Number: </span> {number}
          <br />
          <span className="card-date-time">Date:</span> {date}
          <br />
          <span className="card-location">Location:</span> {site}
          <br />
          <span className="card-location">Rocket Name:</span> {rocketName}
          <br />
          <span className="card-location">Rocket Type:</span> {rocketType}
          <br />
          {/****** TODO: return "no details available" when there are no details 
          this also call a proptype error when details returns null*/}
          <span className="card-location">Mission Details: </span> {details}
          <br />
        </p>

        <div className="links">
          <div className="card-location clearfix">
            <a href={youtube_url} target="_blank" rel="noopener noreferrer">
              <img src={youtubeImg} alt="youtube icon" className="float-left" />
              <span className="youtube">YouTube Video</span>
            </a>
          </div>
          <div className="card-location clearfix">
            <a href={wiki_url} target="_blank" rel="noopener noreferrer">
              <img src={wikiImg} alt="wiki icon" className="float-left" />
              <span className="wiki">Wikipedia Page</span>
            </a>
          </div>
        </div>
      </div>
    </div>
  );
};

export default CardDetails;

CardDetails.propTypes = {
  patch_url: PropTypes.string.isRequired,
  number: PropTypes.number.isRequired,
  name: PropTypes.string.isRequired,
  date: PropTypes.string.isRequired,
  site: PropTypes.string.isRequired,
  rocketName: PropTypes.string.isRequired,
  rocketType: PropTypes.string.isRequired,
  details: PropTypes.string.isRequired,
  youtube_url: PropTypes.string.isRequired,
  wiki_url: PropTypes.string.isRequired
};
