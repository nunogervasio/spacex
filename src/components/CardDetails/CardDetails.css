.main-card-details {
  border-top: solid black 2px;
  border-bottom: solid black 2px;
  padding-top: 16px;
  margin: 0.3em;
  display: flex;
  flex-direction: column;
}
.card-img-details {
  display: flex;
  flex-direction: row;
  justify-content: center;
}

.links {
  display: flex;
  flex-direction: row;
  justify-content: space-around;
  margin-bottom: 1em;
  padding: 1em;
}
.card-details-details {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  margin-left: 1em;
}

.card-location img {
  max-height: 20px;
  max-width: 20px;
}
/* Clearfix (clear floats) */
.clearfix::after {
  content: "";
  clear: both;
  display: table;
}
/****** floated columns */
.youtube,
.wiki {
  float: right;
  margin-left: 5px;
}

.float-left {
  width: 25%;
  float: left;
}
@media (min-width: 769px) {
  .main-card-details {
    flex-direction: row;
  }
  .card-img-details {
    align-self: center;
  }
}
