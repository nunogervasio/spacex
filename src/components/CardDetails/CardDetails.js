import React from "react";
import PropTypes from "prop-types";
import youtubeImg from "../../img/youtube-1.svg";
import wikiImg from "../../img/wiki-1.svg";
import Patch from "../Patch/Patch";
import Links from "../Links/Links";
import MissionDetails from "../MissionDetails/MissionDetails";
import styles from "./CardDetails.module.css";

const CardDetails = ({
  patch_url,
  number,
  name,
  date,
  site,
  rocketName,
  rocketType,
  details,
  youtube_url,
  wiki_url
}) => {
  return (
    <div className={styles.card}>
      <Patch patch_url={patch_url} />
      <div className={styles.details}>
        <MissionDetails
          name={name}
          number={number}
          date={date}
          site={site}
          rocketName={rocketName}
          rocketType={rocketType}
          details={details}
        />
        <Links
          youtube_url={youtube_url}
          youtubeImg={youtubeImg}
          wiki_url={wiki_url}
          wikiImg={wikiImg}
        />
      </div>
    </div>
  );
};

export default CardDetails;

CardDetails.propTypes = {
  patch_url: PropTypes.string.isRequired,
  number: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  date: PropTypes.string.isRequired,
  site: PropTypes.string.isRequired,
  rocketName: PropTypes.string.isRequired,
  rocketType: PropTypes.string.isRequired,
  details: PropTypes.string.isRequired,
  youtube_url: PropTypes.string.isRequired,
  wiki_url: PropTypes.string.isRequired
};
