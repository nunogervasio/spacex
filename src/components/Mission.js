import React from "react";

const Mission = ({ match }) => {
  let name = match.params.name;
  return <h1>Mission is working with {name}!</h1>;
};

export default Mission;
