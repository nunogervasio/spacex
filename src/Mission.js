import React from "react";
import CardDetails from "./components/CardDetails";
import utils from "./utils";

const Mission = props => {
  let name = props.match.params.name;

  //TODO: function that filters out the '/' character of the url path

  let selectedMission = props.data.filter(d => {
    return d.mission_name === name;
  });
  let cards = selectedMission.map((element, index) => {
    return (
      <CardDetails
        key={index}
        patch_url={element.links.mission_patch_small}
        number={utils.addLeadingZero(element.flight_number)}
        name={element.mission_name}
        date={utils.formatDate(element.launch_date_local)}
        site={element.launch_site.site_name_long}
        rocketName={element.rocket.rocket_name}
        rocketType={element.rocket.rocket_type}
        details={element.details}
        core_serial={element.rocket.first_stage.cores[0]}
        youtube={element.links.video_link}
        wiki={element.links.wikipedia}
      />
    );
  });
  return <div className="container">{cards}</div>;
};

export default Mission;
