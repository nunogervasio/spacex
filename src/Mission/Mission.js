import React from "react";
import PropTypes from "prop-types";
import CardDetails from "../components/CardDetails/CardDetails";
import {
  addLeadingZero,
  removeForwardSlash,
  formatDate,
  isDetailsNull
} from "../utils";
import styles from "./Mission.module.css";

const Mission = ({ data, match }) => {
  let name = match.params.name;

  let selectedMission = data.filter(d => {
    return removeForwardSlash(d.mission_name) === name;
  });

  let cards = selectedMission.map((element, index) => {
    return (
      <CardDetails
        key={index}
        patch_url={element.links.mission_patch_small}
        number={addLeadingZero(element.flight_number)}
        name={removeForwardSlash(element.mission_name)}
        date={formatDate(element.launch_date_local)}
        site={element.launch_site.site_name_long}
        rocketName={element.rocket.rocket_name}
        rocketType={element.rocket.rocket_type}
        details={isDetailsNull(element.details)}
        youtube_url={element.links.video_link}
        wiki_url={element.links.wikipedia}
      />
    );
  });

  return <div className={styles.container}>{cards}</div>;
};

export default Mission;

Mission.propTypes = {
  data: PropTypes.array.isRequired,
  match: PropTypes.object.isRequired
};
