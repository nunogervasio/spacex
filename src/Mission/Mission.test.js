import React from "react";
import { shallow } from "enzyme";
import Mission from "./Mission";
import CardDetails from "../components/CardDetails/CardDetails";

describe("Mission", () => {
  const props = {
    data: [],
    match: { params: { name: "" } }
  };
  const mission = shallow(<Mission {...props} />);

  it("renders properly", () => {
    expect(mission).toMatchSnapshot();
  });

  it("creates a <CardDetails /> component", () => {
    // expect(mission.find("CardDetails").exists()).toBe(true);
  });
});
