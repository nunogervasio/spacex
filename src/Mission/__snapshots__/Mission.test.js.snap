// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mission renders properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Mission
    data={Array []}
    match={
      Object {
        "params": Object {
          "name": "",
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [],
      "className": "container",
    },
    "ref": null,
    "rendered": Array [],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [],
        "className": "container",
      },
      "ref": null,
      "rendered": Array [],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
